bpf/process/bpf_generic_lsm_core.c

```diff
@@ -56,6 +56,15 @@ struct {
 	__type(value, struct event_config);
 } config_map SEC(".maps");
 
+#ifdef __LARGE_MAP_KEYS
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, 1024);
+	__type(key, __u64);
+	__type(value, char[65]);
```

Comment:
could we have a struct in here instead of just byte array? and some macro for the size

also I was wondering we should add some state to the hash value that would be increased each time it passed the bpf program in the chain and each program would make sure it is executed on the proper place of the chain

and we'd send some error in the event if it wasn't the case  ... so we would be sure the order of the programs in the trampoline chain is correct (in case it's evenr changed in future) .. wdyt?

Reply:
Nice idea!